control/grpc.py

```diff
@@ -1586,6 +1592,9 @@ def namespace_resize(self, request, context=None):
         nsid_msg = self.get_ns_id_message(request.nsid, request.uuid)
         self.logger.info(f"Received request to resize namespace {nsid_msg}on {request.subsystem_nqn} to {request.new_size} MiB, context: {context}{peer_msg}")
 
+        if request.new_size <= 0:
+            return pb2.req_status(status=errno.EINVAL,
+                                  error_message=f"Failure resizing namespace {nsid_msg}on {request.subsystem_nqn}: New size must be positive")
```

Comment:
Server side resize new size verification
```suggestion
        if request.new_size <= 0:
            return pb2.req_status(status=errno.EINVAL,
                                  error_message=f"Failure resizing namespace {nsid_msg}on {request.subsystem_nqn}: New size must be positive")
        if request.new_sizee % (1024 * 1024):
            return pb2.req_status(status=errno.EINVAL,
                                   error_message=f"Failure resizing namespace {nsid_msg}on {request.subsystem_nqn}: New size must be aligned to MiBs")

                                  
```

Reply:
@baum this is not needed as here, unlike the create, we already pass the size in MiBs, not in bytes. So there is no way it won't be aligned. The SPDK new size parameter is in MiBs, while the RBD API accepts size in bytes.

Reply:
@gbregman good point, it sounds that namespace create gRPC units should be using same image size units as resize.
```
Adhere to the sizing convention used in Ceph, where size-in-M/G/T is specified. If no suffix is provided, the unit M should be assumed.
```

Reply:
@baum this is not in the scope of this issue. I didn't want to change the semantic of the sizes used. If you think we should change that please open a specific issue.